sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/ThresholdBasedAvailabilityStrategy.java

```diff
@@ -0,0 +1,92 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.cosmos;
+
+import java.time.Duration;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.stream.Collectors;
+
+/**
+ * The type Threshold based retry availability strategy.
+ */
+public class ThresholdBasedAvailabilityStrategy extends AvailabilityStrategy{
+    private static final Duration DEFAULT_THRESHOLD_IN_MILLISECONDS = Duration.ofMillis(500);
+    private static final Duration DEFAULT_THRESHOLD_STEP_IN_MILLISECONDS = Duration.ofMillis(100);
+    private Duration threshold;
+    private Duration thresholdStep;
+
+    /**
+     * Instantiates a new Threshold based retry availability strategy.
+     */
+    public ThresholdBasedAvailabilityStrategy() {
+        this.threshold = DEFAULT_THRESHOLD_IN_MILLISECONDS;
+        this.thresholdStep = DEFAULT_THRESHOLD_STEP_IN_MILLISECONDS;
+    }
+
+    /**
+     * Instantiates a new Threshold based retry availability strategy.
+     *
+     * @param threshold            the threshold
+     * @param thresholdStep        the threshold step
+     * @param numberOfRegionsToTry the number of regions to retry
+     */
+    public ThresholdBasedAvailabilityStrategy(Duration threshold, Duration thresholdStep, int numberOfRegionsToTry) {
```

Comment:
do we also need a setter/getter for numberOfRegionsToTry?

Reply:
It comes from the super class